lighthouse-core/config/config.js

```diff
@@ -395,40 +383,59 @@ class Config {
   }
 
   /**
-   * @param {LH.Config} config
-   * @return {LH.Config}
+   * @param {LH.Config.Json['passes']} passes
+   * @return {?Array<Required<LH.Config.PassJson>>}
    */
-  static augmentWithDefaults(config) {
-    const {defaultSettings, defaultPassConfig} = constants;
-    config.settings = merge(deepClone(defaultSettings), config.settings, true);
-    if (config.passes) {
-      config.passes = config.passes.map(pass => merge(deepClone(defaultPassConfig), pass));
+  static augmentPassesWithDefaults(passes) {
+    if (!passes) {
+      return null;
     }
 
-    return config;
+    const {defaultPassConfig} = constants;
+    return passes.map(pass => merge(deepClone(defaultPassConfig), pass));
+  }
+
+  /**
+   * @param {LH.Config.SettingsJson=} settings
+   * @param {LH.Flags=} flags
+   * @return {LH.Config.Settings}
+   */
+  static initSettings(settings = {}, flags) {
+    // Fill in missing settings with defaults
+    const {defaultSettings} = constants;
+    const settingWithDefaults = merge(deepClone(defaultSettings), settings, true);
+
+    // Override any applicable settings with CLI flags
+    const settingsWithFlags = merge(settingWithDefaults || {}, cleanFlagsForSettings(flags), true);
+
+    return settingsWithFlags;
   }
 
   /**
    * Expands the audits from user-specified JSON to an internal audit definition format.
    * @param {LH.Config.Json['audits']} audits
    * @return {?Array<{path: string, options?: {}} | {implementation: typeof Audit, path?: string, options?: {}}>}
    */
-  static expandAuditShorthandAndMergeOptions(audits) {
+  static expandAuditShorthand(audits) {
     if (!audits) {
       return null;
     }
 
     const newAudits = audits.map(audit => {
       if (typeof audit === 'string') {
         return {path: audit, options: {}};
-      } else if (audit && typeof audit.audit === 'function') {
+      } else if ('implementation' in audit && typeof audit.implementation.audit === 'function') {
+        return audit;
```

Comment:
the example comments for gatherer are 💯 want to add them here too?

Reply:
good idea, done